lib/shopify_api/only_sync_dirty_attributes.rb

```diff
@@ -0,0 +1,43 @@
+module ShopifyAPI
+  module OnlySyncDirtyAttributes
```

Comment:
I think it is possible to use `ActiveModel::Dirty` instead of doing a custom implementation.